### ISSUE TITLE
Full error reporting

### ISSUE THREAD

**Issue by singularsyntax** [CESNET/libyang-python#56]
If I use `yanglint` to parse some JSON config input data containing duplicate keyed items, I get an error message like this:

```
$ yanglint -p <yang-schemas> -f json some_schema.yang -t config config.json
libyang err : Duplicate instance of "baz". (Schema location /foo/bar/baz, data location /foo/bar/baz[key='fob'], line number XXX)
YANGLINT[E]: Failed to parse input data file "config.json".
```

Using `libyang-python` API to parse the same data on the same system (libyang version `2.0.194-r0`), (e.g. using `Context.parse_data_mem()`), it throws a `LibyangError` which only includes the first part of the error: `Duplicate instance of "baz"`.

It appears that `libyang-python` attempts to recover the error path information (`Context.error()`) but it's not working for some reason. I wasn't able to see an obvious reason glancing through `libyang` source code given my limited understanding of the C API.

**Comment by rjarry** [comment]
Hi there,

TL;DR: enable python logging callback to see the path information:

```python
libyang.configure_logging(True, logging.ERROR)
```

I assume that the message you are seeing is printed by the log callback:

https://github.com/CESNET/libyang/blob/v2.1.4/tools/lint/main_ni.c#L318

In the python bindings, the logging callback is disabled by default as it can have unwanted side effects and and even cause performance issues (see commit b527e8fe4b80f for details).

It appears that the information you are missing would be part of `err.path` here:

https://github.com/CESNET/libyang-python/blob/v2.5.1/libyang/context.py#L117-L118

After some inspection, it seems that the `path` flag is only enabled when the log callback is set:

https://github.com/CESNET/libyang/blob/v2.1.4/src/log.c#L261-L266

Which causes stored errors to miss that information:

https://github.com/CESNET/libyang/blob/v2.1.4/src/log.c#L661-L663

**Comment by singularsyntax** [comment]
Hi Robin,

Thanks for looking into this and the detailed explanation. Configuring the logging as per above did the trick! Closing this issue.
